lisa/_assets/kmodules/lisa/utils.h

```diff
@@ -12,4 +12,15 @@
 
 #include "linux/kernel.h"
 
+static inline size_t list_count_elements(struct list_head *head)
```

Comment:
Minor size_t is usually used for variables that do represent a size of an object, for counting unsigned int seems more appropriate.
Aside : Is it being used anywhere ? Could not find any reference to it...

Reply:
Drive by comment: size_t is definitely the correct type here. The max size of most collection types is proportional to the amount of addressable memory. sizeof() applied on an array type gives a size_t. Beyond that, the restrictions on `unsigned int` are very loose. A strictly conforming implementation in C is perfectly allowed to have 16 bits `int` type with several MB of RAM. These platforms exist in practice even if the kernel or POSIX is not portable to them. The only thing I can think of that is more of a pain with `size_t` is printing it, since printf has no native support for it, you either use the extra macros or just cast to the largest type you expect and print as that. 

Beyond C, Rust systematically uses `usize` for this case (equivalent to size_t), and sloppiness around that would be paid by extra boilerplate (there is no infallible conversion from `usize` to any other integer type beyond the `as` cast operator that is considered to a large extent as a historical accident).

Reply:
Thanks for the clarification Douglas.
Unfortunately the function doesn't seem to be used anywhere anymore. So I ll remove it.